src/Compilers/VisualBasic/Portable/Parser/ParseExpression.vb

```diff
@@ -1462,6 +1469,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
 
         End Function
 
+        ' After VB15.6 it is possible to use named arguments in non-trailing position, except in attribute lists (where it remains disallowed)
+        Private Shared Function ReportNonTrailingNamedArgumentIfNeeded(argument As ArgumentSyntax, seenNames As Boolean, attributeListParent As Boolean, allowNonTrailingNamedArguments As Boolean) As ArgumentSyntax
+            If seenNames Then
+                If attributeListParent Then
+                    ' PROTOTYPE(non-trailing) should we produce missing tokens (here and in the next case too)?
```

Comment:
Wait. For the attribute case, yes I think we should create missing tokens for name and colonEquals, attach the error to the name token and create new argument node with the tokens.